pkg/genericapiserver/enablement_config.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// APIResourceEnablement is the interface to determine which versions and resources are enabled
+type APIResourceEnablement interface {
+	ResourceEnabled(resource unversioned.GroupVersionResource) bool
+	VersionEnabled(version unversioned.GroupVersion) bool
```

Comment:
This is just a default, since individual resources could still be turned on. I suggest something like
* Change the name to VersionDefaultsOn?
* Change to have both "AllResourcesForVersionEnabled" and "AnyResourcesForVersionEnabled"?